src/documentation/sdks/java-extensions.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: default
+title: "Beam Java SDK Extensions"
+permalink: /documentation/sdks/java-extensions/
+---
+# Apache Beam Java SDK Extensions
+
+## <a name="join-library"></a>Join-library
+
+Join-library provides inner join, outer left and right join functions. The aim
+is to simplify the most common cases of join to a simple function call.
+
+The functions are generic so it supports join of any types supported by
+Beam. Input to the join functions are `PCollections` of `Key` / `Value`s. Both
+the left and right `PCollection`s need the same type for the key. All the join
+functions return a `Key` / `Value` where `Key` is the join key and value is
+a `Key` / `Value` where the key is the left value and right is the value.
+
+In the cases of outer join, since `null` cannot be serialized the user have
+to provide a value that represent `null` for that particular use case.
+
+Example usage:
+
+```java
```

Comment:
the dreaded disappearing language tab problem for blocks on this page too. for Python specific pages, I think right now it just omits the language (so just 3 backticks) so the code block always shows up. So could do that here (though you lose syntax highlighting that way), or add a python tab that has some sort of not applicable note.